src/main/java/com/lms/learning_management_system/entities/service/UserService.java

```diff
@@ -34,7 +34,12 @@ public UserDTO getUserById(UUID id) {
     }
 
     public UserDTO createUser(UserDTO userDTO) {
-        UserEntity userEntity = convertToEntity(userDTO);
+        Optional<UserEntity> existingUser = userRepository.findByEmail(userDTO.getEmail());
```

Comment:
You do a similar operation several times in UserService. For cleaner code, you can make a method and reuse it. Think about it

Reply:
I am not exactly sure how to, i use different stream operation after the findBy and i don't know how to implement that

Reply:
ok leave it at that